analyzers/src/SonarAnalyzer.CSharp/Rules/LoggersShouldBeNamedForEnclosingType.cs

```diff
@@ -62,8 +69,12 @@ private static void Process(SonarSyntaxNodeReportingContext context)
         }
     }
 
-    private static bool IsArgumentInObjectCreation(InvocationExpressionSyntax invocation) =>
-        invocation.Ancestors().OfType<ObjectCreationExpressionSyntax>().Any();
+    private static SyntaxNode EnclosingTypeNode(InvocationExpressionSyntax invocation)
+    {
+        var ancestors = invocation.Ancestors();
+        return (SyntaxNode)ancestors.OfType<ObjectCreationExpressionSyntax>().FirstOrDefault() // prioritize new() over enclosing type
```

Comment:
`SyntaxNode` is needed here because the two sides of `??` are different derivatives of `SyntaxNode`.